### ISSUE TITLE
testing feature request: mock WebGL context for headless browser testing

### ISSUE THREAD

**Issue by brunoNetId** [aframevr/aframe#4608]
**Description:**

For complex scenes with very dynamic functionality there is a need for black box testing. I tried to use mocha/chai/jsdom/canvas to load the HTML A-Frame source, but it fails with the following message:
 - THREE.WebGLRenderer: Error creating WebGL context.

Ideally one should be able, for example, to configure the 'a-scene' renderer passing a 'mock' parameter (e.g. renderer="mock: true")
 reference: https://aframe.io/docs/1.0.0/components/renderer.html

That would allow loading a-frame scenes in a headless web browser and run many end-to-end use cases.

- A-Frame Version: 1.0.4
- Platform / Device: n/a
- Reproducible Code Snippet or URL: n/a

This is what I tried to do (and got the error message from the description):

    let jsdom;
    before(async function() {
            jsdom = await JSDOM.fromFile('under-test.html', {
            resources: "usable",
            runScripts: "dangerously"
        });

        await new Promise(resolve =>
            jsdom.window.addEventListener("load", resolve)
        );
    });

**Comment by dmarcos** [comment]
I think there's nothing actionable on the A-Frame side. We can reopen if necessary

**Comment by djoksimo** [comment]
I just finished setting up some e2e tests for my [Aframe-powered platform](http://scena360.com) that launches chrome instances in "headful" mode and it works in Github actions (using https://github.com/mujo-code/puppeteer-headful); there are forks of the puppeteer-headful action that support other CI systems like GitLab CI/CD too.

I used [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer) and was able to successfully execute Aframe-specific code.

I'm thinking of writing up a blog or something about this. In case you're interested in learning about or implementing e2e tests for your Aframe project, react with a 👍 and I'll look into publishing a blog or distributing some open-source code?
